src/panels/config/integrations/integration-panels/zwave_js/dialog-zwave_js-hard-reset-controller.ts

```diff
@@ -25,9 +25,12 @@ class DialogZWaveJSHardResetController extends LitElement {
 
   public closeDialog(): void {
     this.entry_id = undefined;
-    this._done = false;
 
     fireEvent(this, "dialog-closed", { dialog: this.localName });
+    if (this._done) {
+      this._done = false;
+      setTimeout(() => history.back(), 0);
```

Comment:
You should not do this from the dialog, do it from the caller of the dialog instead.